crates/biome_js_formatter/src/comments.rs

```diff
@@ -624,38 +624,31 @@ fn handle_member_expression_comment(
     }
 }
 
-fn handle_function_declaration_comment(
-    comment: DecoratedComment<JsLanguage>,
-) -> CommentPlacement<JsLanguage> {
-    let is_function_declaration = matches!(
+fn handle_function_comment(comment: DecoratedComment<JsLanguage>) -> CommentPlacement<JsLanguage> {
+    if !matches!(
         comment.enclosing_node().kind(),
         JsSyntaxKind::JS_FUNCTION_DECLARATION
             | JsSyntaxKind::JS_FUNCTION_EXPORT_DEFAULT_DECLARATION
-    );
+            | JsSyntaxKind::JS_FUNCTION_EXPRESSION
+    ) || !comment.kind().is_line()
+    {
+        return CommentPlacement::Default(comment);
+    };
 
-    let following = match comment.following_node() {
-        Some(following) if is_function_declaration => following,
-        _ => return CommentPlacement::Default(comment),
+    let Some(body) = comment.following_node().and_then(JsFunctionBody::cast_ref) else {
+        return CommentPlacement::Default(comment);
     };
 
-    // Make comments between the `)` token and the function body leading comments
-    // of the first non empty statement or dangling comments of the body.
+    // Make line comments between the `)` token and the function body leading comments
+    // of the first statement or dangling comments of the body.
     // ```javascript
-    // function test() /* comment */ {
+    // function test() // comment {
```

Comment:
The JS snippet didn't need to change, this is now **incorrect** because the curly bracket  `{` is part of the comment, unless you meant to **move** the `{` to a new line too,

Reply:
oh yeah, you are right.